### PR TITLE
Hlm 4972

### PR DESCRIPTION
**Pull request by naveenr-egov**
Attendance package with latest flutter version

### CODE REVIEW

File: apps/health_campaign_field_worker_app/lib/blocs/auth/auth.dart [hunk 1]
```diff
@@ -45,6 +52,8 @@ class AuthBloc extends Bloc<AuthEvent, AuthState> {
       final refreshToken = await localSecureStore.refreshToken;
       final userObject = await localSecureStore.userRequestModel;
       final actionsList = await localSecureStore.savedActions;
+
+      final loggedInIndividual = await localSecureStore.userIndividualId;
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/blocs/auth/auth.dart:56]: change the naming convention

File: apps/health_campaign_field_worker_app/lib/blocs/search_households/search_households.dart [hunk 2]
```diff
@@ -165,7 +166,6 @@ class SearchHouseholdsBloc
         clientReferenceId:
             beneficiaries.map((e) => e.beneficiaryClientReferenceId!).toList(),
       ));
-      print(households.length);
     } else {
       individuals = await individual.search(
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/blocs/search_households/search_households.dart:170]: reason for empty search query

File: apps/health_campaign_field_worker_app/lib/data/local_store/sql_store/sql_store.dart [hunk 3]
```diff
@@ -41,18 +46,21 @@ import 'tables/service.dart';
...
-import 'tables/h_f_referral.dart';
 
 part 'sql_store.g.dart';
 
 @DriftDatabase(tables: [
-  Address,
+  AttendanceRegister,
+  Attendance,
+  Attendee,
+  at.Address, // TODO: address same in sql_store.g.dart and rename the address class created in the same file to avoid conflict
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/data/local_store/sql_store/sql_store.dart:63]: Is this been added then please remove the TODO

File: apps/health_campaign_field_worker_app/lib/data/repositories/sync/sync_up.dart [hunk 4]
```diff
@@ -108,6 +108,12 @@ class PerformSyncUp {
                 );
               }
 
+              // if (updatedEntity is HCMAttendanceLogModel) {
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/data/repositories/sync/sync_up.dart:111]: remove the un-used code

File: apps/health_campaign_field_worker_app/lib/models/auth/auth_model.dart [hunk 5]
```diff
@@ -54,8 +54,9 @@ class UserRequestModel with _$UserRequestModel {
 
 @freezed
 class UserRoleModel with _$UserRoleModel {
+  //{TODO: Need to make it optional
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/models/auth/auth_model.dart:57]: Please remove the TODO comments once it's done

File: apps/health_campaign_field_worker_app/lib/blocs/localization/localization.dart
Comment (naveenr-egov) [inline apps/health_campaign_field_worker_app/lib/blocs/localization/localization.dart:1]: remove unused import

File: packages/attendance_management/lib/widgets/attendance_acknowledgement.dart [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: [
+                  Text(
+                    textAlign: TextAlign.center,
+                    localizations.translate(
+                      i18.attendance.attendanceSubmittedSuccessMsg,
+                    ),
+                    style: TextStyle(
+                      fontSize: 32,
```
Comment (naveen-egov) [inline packages/attendance_management/lib/widgets/attendance_acknowledgement.dart:87]: This font size should be in digit themes

File: packages/attendance_management/lib/widgets/attendance_acknowledgement.dart [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+                              color: Theme.of(context).colorScheme.secondary,
+                            ),
+                            minimumSize: Size(
+                              MediaQuery.of(context).size.width / 1,
+                              50,
+                            ),
+                            shape: null,
+                          ),
+                          label: widget.secondaryLabel ?? "Back",
+                          onPressed: widget.secondaryAction,
```
Comment (naveen-egov) [inline packages/attendance_management/lib/widgets/attendance_acknowledgement.dart:147]: remove the hardcoded text and move to constants

File: packages/attendance_management/lib/widgets/circular_button.dart [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+                    padding: const EdgeInsets.only(
+                      left: 4.0,
+                    ),
+                    child: Text(
+                      AttendanceLocalization.of(context)
+                          .translate(i18.attendance.halfDay),
+                      style: DigitTheme
+                          .instance.mobileTheme.textTheme.labelSmall
+                          ?.apply(
+                        color: const Color.fromRGBO(244, 169, 56, 1),
```
Comment (naveen-egov) [inline packages/attendance_management/lib/widgets/circular_button.dart:59]: Move to digit theme

File: packages/attendance_management/lib/widgets/circular_button.dart [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+                  Container(
+                    height: 30,
+                    width: 30,
+                    margin: const EdgeInsets.all(2.0),
+                    decoration: BoxDecoration(
+                      shape: BoxShape.circle,
+                      color: index.isNegative || index == 0.0
+                          ? Colors.white
+                          : index == 0.5
+                              ? const Color.fromRGBO(244, 169, 56, 1)
```
Comment (naveen-egov) [inline packages/attendance_management/lib/widgets/circular_button.dart:76]: use digit theme

File: packages/attendance_management/lib/widgets/no_result_card.dart [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+
+    return Align(
+      alignment: align,
+      child: Padding(
+        padding: const EdgeInsets.only(top: 100, bottom: 10, right: 8),
```
Comment (naveen-egov) [inline packages/attendance_management/lib/widgets/no_result_card.dart:22]: move to constants

**Review by naveen-egov — CHANGES_REQUESTED** [review]

**Review by naveen-egov — APPROVED** [review]

**Pull request merged**